[vulkan][CI] CI android build abi x86 with USE_VULKAN

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39767 [vulkan][CI] CI android build abi x86 with USE_VULKAN**
* #39771 [vulkan][build_fix] Fix Vulkan Build; Prepacking uses new register api

Adding android build for every PR with `-DUSE_VULKAN=ON` It will use vulkan from ANDROID_NDK, so no changes for docker images.

Differential Revision: [D21976091](https://our.internmc.facebook.com/intern/diff/D21976091)

IvanKobzarev committed:
[vulkan][CI] CI android build abi x86 with USE_VULKAN

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan][CI] CI android build abi x86 with USE_VULKAN"




[ghstack-poisoned]

ezyang reviewed (approved):
Should this run by default on PRs or not?

IvanKobzarev committed:
Update on "[vulkan][CI] CI android build abi x86 with USE_VULKAN"


Adding android build for every PR with `-DUSE_VULKAN=ON` It will use vulkan from ANDROID_NDK, so no changes for docker images.

Differential Revision: [D21976091](https://our.internmc.facebook.com/intern/diff/D21976091)

[ghstack-poisoned]

zou3519 commented:
ci/circleci: pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build seems to be flaking out ever since this PR over "vulcan" build failures. See https://circleci.com/gh/pytorch/pytorch/5789135?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link as an example and observe the flakiness in https://ezyang.github.io/pytorch-ci-hud/build2/pytorch-master.

Reading through the contents of the diff, it looks like it adds CI job named "pytorch_linux_xenial_py3_clang5_android_ndk_r19c_x86_32_vulkan_build", but the job that is flaking out is "pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build", so the problem might not actually be here.

However, in an abundance of caution I am going to revert this diff.

IvanKobzarev commented:
> @IvanKobzarev
> ci/circleci: pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build seems to be flaking out ever since this PR over "vulcan" build failures. See https://circleci.com/gh/pytorch/pytorch/5789135?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link as an example and observe the flakiness in https://ezyang.github.io/pytorch-ci-hud/build2/pytorch-master.
> 
> Reading through the contents of the diff, it looks like it adds CI job named "pytorch_linux_xenial_py3_clang5_android_ndk_r19c_x86_32_vulkan_build", but the job that is flaking out is "pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build", so the problem might not actually be here.
> 
> However, in an abundance of caution I am going to revert this diff.

I think that they collided on the same docker image
